File: src/main/kotlin/com/poc/spring/security/oauth2/server/rest/UserController.java
package com.poc.spring.security.oauth2.server.rest;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.security.Principal;

@RestController
public class UserController {

    @GetMapping("/user/me")
    public Principal user(Principal principal) {
        return principal;
    }
}

File: src/main/kotlin/com/poc/spring/security/oauth2/server/userprofile/UserController.kt
import org.springframework.http.ResponseEntity
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.security.core.userdetails.User
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController


@RestController
class UserController {

    @GetMapping("/oauth/user/me")
    fun profile(): ResponseEntity<UserProfile> {
        //Build some dummy data to return for testing
        val user: User = SecurityContextHolder.getContext().authentication.principal as User

        val profile = UserProfile()
        profile.name = user.username

        return ResponseEntity.ok(profile)
    }
}
